extensions/amp-experiment/0.1/amp-experiment.js

```diff
@@ -44,7 +44,13 @@ export class AmpExperiment extends AMP.BaseElement {
     const config = this.getConfig_();
     const results = Object.create(null);
     const variants = Object.keys(config).map(experimentName => {
-      return allocateVariant(this.getWin(), config[experimentName])
+      assertName(experimentName);
+      const experimentConfig = config[experimentName];
+      // If no 'grouping' is specified, fallback to experiment name.
+      if (!experimentConfig.grouping) {
```

Comment:
I agree immutable object is preferred. Any 'side effect' to the param passed in would sometimes surprise the caller.

Here, alternatively we can clone a new object. 
I chose to pass the `experimentName` because in the next PR I will need the name for other purpose (variant override via URL)